File: solutions/1603-running-sum-of-1d-array/solution.rs
impl Solution {
    pub fn running_sum(nums: Vec<i32>) -> Vec<i32> {
        let mut result = Vec::with_capacity(nums.len());
        let mut sum = 0;

        for i in nums{
            sum += i;
            result.push(sum);
        }

        result
    }
}

File: solutions/1603-running-sum-of-1d-array/solution.py
class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        running_sum = [0] * len(nums)
        s = 0
        for i in range(len(nums)):
            s += nums[i]
            running_sum[i] = s
        return running_sum 
